examples/exex-test/src/main.rs

```diff
@@ -0,0 +1,132 @@
+use futures_util::StreamExt;
+use reth::{api::FullNodeComponents, builder::NodeTypes, primitives::EthPrimitives};
+use reth_chainspec::{ChainSpecBuilder, MAINNET};
+use reth_e2e_test_utils::testsuite::{
+    actions::ProduceBlocks,
+    setup::{NetworkSetup, Setup},
+    TestBuilder,
+};
+use reth_exex::{ExExContext, ExExEvent};
+use reth_node_ethereum::{EthEngineTypes, EthereumNode};
+use std::sync::{
+    atomic::{AtomicBool, AtomicU64, Ordering},
+    Arc,
+};
+mod wal_test;
+
+#[allow(dead_code)]
+struct TestState {
+    received_blocks: AtomicU64,
+    saw_trie_updates: AtomicBool,
+    last_finalized_block: AtomicU64,
+}
+
+/// ExEx that tests assertions about notifications and state
+#[allow(dead_code)]
+async fn test_assertion_exex<
+    Node: FullNodeComponents<Types: NodeTypes<Primitives = EthPrimitives>>,
+>(
+    mut ctx: ExExContext<Node>,
+) -> eyre::Result<()> {
+    let state = Arc::new(TestState {
+        received_blocks: AtomicU64::new(0),
+        saw_trie_updates: AtomicBool::new(false),
+        last_finalized_block: AtomicU64::new(0),
+    });
+
+    println!("Assertion ExEx started");
+
+    // Clone state for the async block
+    let state_clone = state.clone();
+
+    // Process notifications
+    while let Some(result) = ctx.notifications.next().await {
+        // Handle the Result with ?
+        let notification = result?;
+
+        // Check for committed chain
+        if let Some(committed_chain) = notification.committed_chain() {
+            let range = committed_chain.range();
+            let blocks_count = *range.end() - *range.start() + 1;
+            println!("Received committed chain: {:?}", range);
+
+            // Increment blocks count
+            #[allow(clippy::unnecessary_cast)]
+            state_clone.received_blocks.fetch_add(blocks_count as u64, Ordering::SeqCst);
+
+            // Send event that we've processed this height
+            ctx.events.send(ExExEvent::FinishedHeight(committed_chain.tip().num_hash()))?;
+
+            // Check for finalization
+            state_clone.last_finalized_block.store(committed_chain.tip().number, Ordering::SeqCst);
+        }
+
+        // For example, if we see any block, we'll set saw_trie_updates to true
+        // This is a simplification
+        state_clone.saw_trie_updates.store(true, Ordering::SeqCst);
+    }
+
+    // Report results at the end
+    report_test_results(&state);
+
+    Ok(())
+}
+
+/// Verify test assertions after completion
+#[allow(dead_code)]
```

Comment:
```suggestion
#[expect(dead_code)]
```
let's use `expect` everywhere instead of `allow`, it helps us maintain the code since we are warned then when the lint expectation is not more fulfilled

Reply:
I have implemented
 ``` #[allow(dead_code)]``` to be ``` #[expect(dead_code)]``` 
  ``` #[allow(dead_code)]``` to be ``` #[allow(unfulfilled_lint_expectations)]``` 
 acc to ```make pr``` result